src/lxc/log.c

```diff
@@ -222,7 +222,7 @@ int lxc_unix_epoch_to_utc(char *buf, size_t bufsize, const struct timespec *time
 	seconds = (time->tv_sec - d_in_s - h_in_s - (minutes * 60));
 
 	/* Make string from nanoseconds. */
-	ret = snprintf(nanosec, LXC_NUMSTRLEN64, "%ld", time->tv_nsec);
+	ret = snprintf(nanosec, LXC_NUMSTRLEN64, "%ld", (long)time->tv_nsec);
```

Comment:
Hm. I'd rather prefer if we'd use `uin64_t` here so:

```
ret = snprintf(nanosec, LXC_NUMSTRLEN64, "%"PRId64, (int64_t)time->tv_nsec);
```

Reply:
Unlike tv->sec (Y2038), tv->nsec is guaranteed to never exceed 10⁹: it'd roll over to a whole second, which is explicitly forbidden — the kernel checks for and rejects such values on input, and won't produce them itself.  Thus, the ABI using a 64-bit field on some archs here is a bit pointless (I guess, done for alignment reasons?).

But on the other hand, using a too wide field is harmless, and makes it easier to avoid mistakes with time_t next to it.

Reply:
I make it a policy to cast to stdint types and usually the maximum one when it can be safely done. This should put us on the safe side on most platforms.